GVFS/GVFS.Virtualization/FileSystemCallbacks.cs

```diff
@@ -197,11 +198,32 @@ public void Stop()
             this.stopping = true;
             lock (this.postFetchJobLock)
             {
-                // TODO(Mac): System.PlatformNotSupportedException: Thread abort is not supported on this platform
-
-                if (!GVFSPlatform.Instance.IsUnderConstruction)
+                if (this.backgroundGitProcess != null)
                 {
-                    this.postFetchJobThread?.Abort();
+                    try
+                    {
+                        if (this.backgroundGitProcess.KillRunningProcess())
+                        {
+                            this.context.Tracer.RelatedEvent(
+                                EventLevel.Informational,
+                                PostFetchTelemetryKey + ": killed background Git process during " + nameof(this.Stop),
+                                metadata: null);
+                        }
+                        else
+                        {
+                            this.context.Tracer.RelatedEvent(
+                                EventLevel.Informational,
+                                PostFetchTelemetryKey + ": failed to kill background Git process during " + nameof(this.Stop),
+                                metadata: null);
+                        }
+                    }
+                    catch (NullReferenceException)

```

Comment:
This catch statement is too broad because of the amount of code inside the try block. E.g. we've had bugs before where a tracer object was null, and this would hide that bug.

I think you can pretty easily avoid having to catch `NullReferenceException` here using an interlocked compare/exchange.

Reply:
in this case, we can simply create a local variable and use plain assignment, then refer only to the local copy. A similar pattern can occur in `TryKillRunningProcess()`.